### PR TITLE
TechDocs: Remove techdocs-cli package from mono-repo

### PR DESCRIPTION
**Pull request by OrkoHunter**
New home: https://github.com/backstage/techdocs-cli

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
🎉

**Comment by OrkoHunter** [comment]
Rebasing after #3230 is merged, will fix the CI for this PR.

**Review by hooloovooo — APPROVED** [review]

File: docs/support/project-structure.md [hunk 1]
```diff
@@ -168,8 +168,8 @@ are separated out into their own folder, see further down.
   package. The Backstage Storybook is found
   [here](https://backstage.io/storybook)
 
-- [`techdocs-cli/`](https://github.com/backstage/backstage/tree/master/packages/techdocs-cli) -
-  Used for verifying TechDocs locally.
+- [`techdocs-cli/`](https://github.com/backstage/techdocs-cli) - Used for
+  verifying TechDocs locally.
```
Comment (freben) [inline docs/support/project-structure.md:172]: this line should be removed entirely shouldn't it - it's no longer part of the project stucture, at least not `packages` (which is the headline)

Comment (OrkoHunter) [reply docs/support/project-structure.md:172]: Yes @freben, you're right. Will remove. :)

**Comment by freben** [comment]
This needs rebase now.

**Comment by OrkoHunter** [comment]
Thanks @freben, done!

**Review by freben — APPROVED** [review]

**Pull request merged**